### DISCUSSION TITLE
Blank values for hidden fields on form submission

### DISCUSSION THREAD

**Discussion by wolispace** [solspace/craft-freeform#1569] [Feature Requests]
We have an Email field for the user to enter their email address in.  

Its visibility is controlled by a radio button on the form "Do you want to be contacted"

If the user entered a dodgy email address eg "NA" then toggles the visibility of the Email field OFF:

1) It passed initial validation as its not visible
2) trying to send the email notification fails and we get an admin alert:  Email "NA" does not comply with addr-spec of RFC 2822.

Is there a way of blanking the Email field if its not visible when the Submit button is pressed?

The only Rules I can see toggle visibility not content.

Craft Pro 4.12.5 
Freeform 5.6.3

**Comment by gustavs-gutmanis** [comment]
Hi @wolispace 

We're using [this library](https://github.com/egulias/EmailValidator) to validate emails, as well as a check for `@` and a `.` in the domain part of the email.

All of the validations happen in the backend, so it can't be bypassed in the front-end. Could you describe how your Email Field is set up, please? And is it possible for you to reproduce the same `NA` value when filling out the form?

Edit:

I see that it might be caused by a person filling in the email field and then having a rule hiding it - in that case, the validation is skipped. We will look into this issue, to make sure the notifications aren't triggered if the field has been hidden by rules.

**Reply by wolispace** [reply]
Yes, thats right, the email is incorrectly filled in, then a rule is applied that hides the email (use selectes a lightwitch to bot be contacted), but the validation is still applied on submission.

Two options I can think of:
1) blank all values when a rule to hide any field is applied.
2) check the visibility, at submit time, of all fields before applying validation.

Thanks

**Comment by kjmartens** [comment]
This should now be resolved in Freeform 5.9. 🙂
